pkg/watch/watch.go

```diff
@@ -320,25 +324,81 @@ func (o *WatchClient) eventWatcher(ctx context.Context, sourcesWatcher *fsnotify
 
 	podsPhases := NewPodPhases()
 
+	// Open up a tty to read in character keys from the user, this will be done in the background
```

Comment:
Wouldn't we have to keep it in the eventWatcher function because that's where the main select loop is? I had it incorporated in WatchAndPush before until I realized that we have to have the main loop in order to coordinate full pushing / updating.

Reply:
To stick with how it is done for other watchers, you would need to have a `createKeyWatcher` function similar to :

```
func createKeyWatcher() (chan string {
        manualSync := make(chan string)
        // In order to capture the keyboard events, we'll use the "tty" library.
	// Open up a tty to read in character keys from the user, this will be done in the background
	tty, err := tty.Open()
	if err != nil {
		klog.Fatal(err)
	}
	defer tty.Close()

	// Create a go routine that will read a character from the tty in the background without interruption
	go func() {
		for {
			char, err := tty.ReadRune()
			if err != nil {
				klog.Fatal(err)
			}
			// If we received the "r" character, we will push to the open channel for manual syncing.
			if string(char) == "p" {
				manualSync <- string(char)
			}
		}
	}()
        return manualSync
}
```

and call this function from `watchAndPush`, then pass the returned `manualSync` to `eventWatcher`

This has been done this way (creating the channels outside of the `eventWatcher` function and passing them to `eventWatcher`), to be able to unit-test the `eventWatcher` by faking the channels.

Reply:
Ahhhh, I see. I've gone ahead and implemented that. Working on tests now.